### ISSUE TITLE
Images can't be loaded if resized to very small sizes

### ISSUE THREAD

**Issue by grzesiek2010** [getodk/collect#5033]
#### Problem description
Images can't be loaded if resized to very small sizes.

#### Steps to reproduce the problem
The user claims that he is able to reproduce the issue selecting `Very small (640px)` option from `Settings` -> `Form management` -> `Image size`.  I've tried it on like 4 devices to no avail.

#### Other information 
The issue has been originally reported on the forum: https://forum.getodk.org/t/image-problem-for-very-small-640px-selected-file-is-not-a-valid-image-on-v2022-1-1-when-resized/36707

**Comment by grzesiek2010** [comment]
Finally I was able to reproduce the issue. It occurs only on some devices (it depend on installed camera app) and only if we scale images down using small values (it also depends I was only able to reproduce the issue using smaller values like 200px).
The problem is more common after merging https://github.com/getodk/collect/pull/5004 but it's not caused by that pull request.
[It is caused by saving exif data to a file that we previously scaled down](https://github.com/getodk/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/utilities/ImageConverter.kt#L48)
From my investigation I can say it happens when exif contains a thumbnail and the image we produced (scaled down image) is small. Then an exception is thrown which we catch but it corrupts our images and we can't use it.

The fact that we copy exif data from an original image and add it to our edited one is a mistake. Apart from that problem with thumbnail exif data contains also attributes like image width and height. Then those attributes still hold values from the original image what doesn't make sense.
My proposal would be to just pick some essential attributes like:
- TAG_GPS_LATITUDE
- TAG_GPS_LONGITUDE
- TAG_GPS_ALTITUDE
- TAG_DATETIME

and maybe some more and copy them to the new image ignoring everything else that might be not valid for the new image.

@seadowg @lognaturel what do you think?

**Comment by lognaturel** [comment]
Can we do it the other way around and remove the attributes that cause problems, keeping the rest?

**Comment by grzesiek2010** [comment]
It is not possible because that thumbnail is not a normal attribute that I can change or remove. I have no access to it. Alternatively I can try creating a new exif (so that it doesn't contain any thumbnail) and copying exif attributes from the old one to the new one.

**Comment by grzesiek2010** [comment]
It is not possible because that thumbnail is not a normal attribute that I can change or remove. I have no access to it. Alternatively I can try creating a new exif (so that it doesn't contain any thumbnail) and copying exif attributes from the old one to the new one.

**Comment by lognaturel** [comment]
> thumbnail is not a normal attribute

Of course it isn't. ARG! 

> try creating a new exif

That sounds promising to me. My reasoning is that if we try to maintain an "ok list" of attributes that we do want we'll inevitably remove something weird that a user wants. We're more likely to be able to keep up with a "no list" of attributes we don't want maintained.

**Comment by grzesiek2010** [comment]
I'm thinking about something like this: https://github.com/getodk/collect/pull/5035
it contains the list of attributes (it's not available in ExifInterface to access) so I copied all tags and created a list that then I navigate through copying all attributes. What do you think?

**Comment by lognaturel** [comment]
That’s the “ok list” I described above. I was proposing you have a list of the few we DON’T want, iterate through all the keys and copy unless it’s in the no list. That will be much easier to read and maintain. Is there a way to avoid the thumbnail that way?

**Comment by grzesiek2010** [comment]
> iterate through all the keys

That's what I did. I iterate through all the keys but to do that I had to build my own list of all tags because `ExifInterface` does not provide any list like that...
I tested older versions of `androidx.exifinterface` and if we downgrade it looks like I can't reproduce the issue. Maybe for now we should just downgrade and report the issue https://issuetracker.google.com/ I don't know if it's a bug or not but maybe they will explain.
